cmd/flux/events.go

```diff
@@ -103,21 +111,33 @@ func eventsCmdRun(cmd *cobra.Command, args []string) error {
 	}
 
 	var diffRefNs bool
-	clientListOpts := getListOpt(namespace, eventArgs.forSelector)
+	clientListOpts := []client.ListOption{client.InNamespace(*kubeconfigArgs.Namespace)}
+	if err != nil {
+		return err
+	}
 	var refListOpts [][]client.ListOption
 	if eventArgs.forSelector != "" {
-		refs, err := getObjectRef(ctx, kubeclient, eventArgs.forSelector, *kubeconfigArgs.Namespace)
+		kind, name := utils.ParseObjectKindName(eventArgs.forSelector)
+		if kind == "" || name == "" {
+			return fmt.Errorf("--for selector must be of format <kind>/<name>")
```

Comment:
Shouldn't this rather be like this?

```suggestion
		if kind == "" {
			return fmt.Errorf("--for selector must be of format <kind>[/<name>]")
```

Reply:
we follow kubectl event's `--for` flag and only support  `<kind>/<name>` .
But it might a good addition in a follow up pr

Reply:
Okay. Updated it so it does this. I think it's a great addition. Thanks!